autogpt_platform/backend/backend/blocks/example/example.py

```diff
@@ -0,0 +1,143 @@
+                        {"message": "Hello, world!"},
+                        {"message": "Hello, world!"},
+                    ],
+                ),
+                ("greeting_count", 3),
+            ],
+            # Function names on the block implementation to mock on test run.
+            # Each mock is a dictionary with function names as keys and mock implementations as values.
+            test_mock={
+                "my_function_that_can_be_mocked": lambda *args, **kwargs: "Hello, world!"
+            },
+            # The credentials required for testing the block.
+            # This is an instance of APIKeyCredentials with sample values.
+            test_credentials=TEST_CREDENTIALS,
+            # The type of the block, which is an instance of BlockType Enum.
+            block_type=BlockType.STANDARD,
+        )
+
+    @staticmethod
+    def my_static_method(post_greeting: str) -> str:
+        logger.info("my_static_method called with input: %s", post_greeting)
+        return f"Hello, {post_greeting}!"
+
+    def my_function_that_can_be_mocked(
+        self, name: str, credentials: APIKeyCredentials
+    ) -> str:
+        logger.info("my_function_that_can_be_mocked called with input: %s", name)
+
+        # Use the ExampleClient from _api.py to make an API call
+        client = ExampleClient(credentials=credentials)
+
+        # Create a sample resource using the client
+        resource_data = {"name": name, "type": "greeting"}
+        response = client.create_resource(resource_data)
+        return f"API response: {response.get('message', 'Hello, world!')}"
+
+    def run(
+        self, input_data: Input, *, credentials: APIKeyCredentials, **kwargs
+    ) -> BlockOutput:
+        """
+        The run function implements the block's core logic. It processes the input_data
+        and yields the block's output.
+
+        In addition to credentials, the following parameters can be specified:
+            graph_id: The ID of the graph containing this block.
+            node_id: The ID of this block's node in the graph.
+            graph_exec_id: The ID of the current graph execution.
+            node_exec_id: The ID of the current node execution.
+            user_id: The ID of the user executing the block.
+        """
+        rtn_all_responses: list[dict[str, Any]] = []
+        # Here we deomonstrate best practice for blocks that need to yield multiple items.
+        # We yield each item from the list to allow for operations on each element.
+        # We also yield the complete list for situations when the full list is needed.
+        for greeting in input_data.greetings:
+            full_greeting = ExampleBlock.my_static_method(greeting)
+            message = self.my_function_that_can_be_mocked(full_greeting, credentials)
+            rtn_all_responses.append({"message": message})
+            yield "response", {"message": message}
+        yield "all_responses", rtn_all_responses
```

Comment:
I don't know if this is a good pattern to share with others. It will pollute the output & replicate data unnecessarily. 

@ntindle @Torantulino I know you've been using this approach for batching the responses, is there a way to get around this to be handled natively?

Reply:
I don't care about the count but yielding the group and individual is super useful for QoL